validation/validation.go

```diff
@@ -29,34 +28,25 @@ func (r *Validation) Make(data any, rules map[string]string, options ...validate
 		return nil, errors.New("rules can't be empty")
 	}
 
-	var dataType reflect.Kind
-	switch data := data.(type) {
+	var dataFace validate.DataFace
+	var err error
+	switch td := data.(type) {
+	case validate.DataFace:
+		dataFace = td
 	case map[string]any:
-		if len(data) == 0 {
+		if len(td) == 0 {
 			return nil, errors.New("data can't be empty")
 		}
-		dataType = reflect.Map
-	}
-
-	val := reflect.ValueOf(data)
-	indirectVal := reflect.Indirect(val)
-	typ := indirectVal.Type()
-	if indirectVal.Kind() == reflect.Struct && typ != reflect.TypeOf(time.Time{}) {
-		dataType = reflect.Struct
-	}
-
-	var dataFace validate.DataFace
-	switch dataType {
-	case reflect.Map:
-		dataFace = validate.FromMap(data.(map[string]any))
-	case reflect.Struct:
-		var err error
+		dataFace = validate.FromMap(td)
+	case url.Values:
+		dataFace = validate.FromURLValues(td)
+	case map[string][]string:
```

Comment:
> Ditto, do we have a situation that uses this case?

We don't use it directly at the moment, but `Make` method accepts the any type, and I think it can be added.